src/test/java/org/folio/circulation/resources/renewal/RegularRenewalStrategyTest.java

```diff
@@ -0,0 +1,224 @@
+        "period for holds is specified")));
+  }
+
+  @Test
+  public void cannotRenewWhenHoldRequestedAndFixedPolicyHasRenewPeriod() {
+    final var request = new RequestBuilder().hold().asDomainObject();
+    final var loanPolicy = new LoanPolicyBuilder()
+      .fixed(UUID.randomUUID())
+      .renewWith(days(10))
+      .withHolds(null, true, null)
+      .asDomainObject();
+
+    final var renewResult = renew(loanPolicy, request);
+
+    assertThat(renewResult, hasValidationError(
+      hasMessage("Item's loan policy has fixed profile but renewal period is specified")));
+  }
+
+  @Test
+  @Parameters({
+    "Declared lost",
+    "Aged to lost",
+    "Claimed returned",
+  })
+  public void cannotRenewItemsWithDisallowedStatuses(String itemStatus) {
+    final var loanPolicy = new LoanPolicyBuilder().asDomainObject();
+    final var loan = new LoanBuilder().asDomainObject()
+      .withLoanPolicy(loanPolicy)
+      .changeItemStatusForItemAndLoan(ItemStatus.from(itemStatus));
+
+    final var renewResult = renew(loan, null);
+
+    assertThat(renewResult, hasValidationError(hasMessage("item is " + itemStatus)));
+  }
+
+  @Test
+  public void cannotRenewLoanThatReachedRenewalLimit() {
+    final var renewalLimit = 2;
+    final var loanPolicy = new LoanPolicyBuilder()
+      .withRenewable(true).limitedRenewals(renewalLimit).asDomainObject();
+    final var loan = Loan.from(new JsonObject().put("renewalCount", renewalLimit + 1))
+      .withLoanPolicy(loanPolicy);
+
+    final var renewResult = renew(loan, null);
+
+    assertThat(renewResult, hasValidationError(hasMessage("loan at maximum renewal number")));
+  }
+
+  @Test
+  public void cannotRenewWhenDueDateCannotBeCalculated() {
+    final var loanPolicy = new LoanPolicyBuilder().rolling(days(10))
+      .withRenewFrom("INVALID_RENEW_FROM").asDomainObject();
+
+    final var renewResult = renew(loanPolicy);
+
+    assertThat(renewResult, hasValidationError(hasMessage("cannot determine when to renew from")));
+  }
+
+  @Test
+  public void cannotRenewWhenPolicyDueDateIsEarlierThanCurrentDueDate() {
+    final var rollingPeriod = days(11);
+    final var loanPolicy = new LoanPolicyBuilder()
+      .rolling(rollingPeriod)
+      .renewFromSystemDate()
+      .asDomainObject();
+
+    final var loan = new LoanBuilder().asDomainObject()
+      .changeDueDate(now(UTC).plusMinutes(rollingPeriod.toMinutes() * 2))
+      .withLoanPolicy(loanPolicy);
+
+    final var renewResult = renew(loan, null);
+
+    assertThat(renewResult, hasValidationError(hasMessage("renewal would not change the due date")));
+  }
+
+  private Result<Loan> renew(Loan loan, Request topRequest) {
+    final List<Request> queueList = topRequest != null ? singletonList(topRequest)
+      : emptyList();
+    final var requestQueue = new RequestQueue(queueList);
+    final var systemDate = now(UTC);
+
+    return new RegularRenewalStrategy().renew(loan, systemDate, requestQueue);
+  }
```

Comment:
I'd prefer to avoid branching (especially on magic null values) in tests if possible, I think it makes them more difficult to follow. 

What do you think about two methods?

And if you want, a renew(loan, queue) method for the commonality.

```suggestion
  private Result<Loan> renew(Loan loan, Request topRequest) {
    val requestQueue = new RequestQueue(singletonList(topRequest));
    val systemDate = now(UTC);

    return new RegularRenewalStrategy().renew(loan, systemDate, requestQueue);
  }
  
  private Result<Loan> renew(Loan loan) {
    val requestQueue = new RequestQueue(emptyList());
    val systemDate = now(UTC);

    return new RegularRenewalStrategy().renew(loan, systemDate, requestQueue);
  }
```